lib/Dialects/VectorExt/Transform/VisitorUtils.h

```diff
@@ -0,0 +1,168 @@
+// Note: This file is mostly a copy-paste of
+// tensorflow/compiler/mlir/tensorflow/utils/visitor_util.h. We copied it here
+// for experimentation purposes within the VectorExt dialect. Please, do not
+// create a dependency with it outside of the VectorExt dialect since it will go
+// away.
+
+//===- VisitorUtils.h -------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef IREE_LLVM_SANDBOX_DIALECT_VECTOREXT_TRANSFORM_VISITOR_UTIL
+#define IREE_LLVM_SANDBOX_DIALECT_VECTOREXT_TRANSFORM_VISITOR_UTIL
+
+#include "mlir/IR/Visitors.h"
+#include <utility>
+
+// This file defines generic (pre/in/post)-order MLIR IR visitors/walkers. The
+// walk() utility that MLIR core provides traverses operations in a block/
+// blocks in a region in the program order, and these walkers do the same. When
+// operations have regions attached to them, the core MLIR walkers visit the
+// regions attached to an Op first, and then visit the op. So within the context
+// of a single Op, the traversal is post-order (considering the Op as the parent
+// node and regions as the children). For certain use cases, it may be more
+// efficient/desirable to visit the parent Op before visiting the attached
```

Comment:
The need for such infra resurfaces time and again.
Can we split this in a proper independent PR with its own tests + test lib that will essentially print the name of ops traversed and cover all the cases of interest?

Reply:
Same as before: Note that the genericWalk code is a copy-paste from the TF repo (see comment at the top VisitorUtils.cpp/h). I think Rahul Joshi (its owner) had plans to upstream this to core.

Reply:
ah I missed that sorry, LGTM to land then and we'll iterate towards a good solution with core and reuse.